### ISSUE TITLE
Update documentation about `hub.config`

### ISSUE THREAD

**Issue by consideRatio** [jupyterhub/zero-to-jupyterhub-k8s#3271]
I think the for `hub.config` should be refined to:

- list exceptions when using it causes problem, like `hub.baseUrl`
- warn or error if known exceptions are configured via `NOTES.txt`
- suggest that its fine to use for other things except for the exception

## Context

[values.schema.yaml entry defining the docs for `hub.config`](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/2f74f28ec83ff54418bdbcc06917248eecfb26ff/jupyterhub/values.schema.yaml#L208-L273)

<details>
<summary>`hub.config` reference docs rendered</summary>

![image](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/assets/3837114/e443ba7c-abd9-4624-b81f-acb36b571153)

</details>

**Comment by consideRatio** [comment]
From @manics in https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/3266#issuecomment-1824338932

> I think we can either say
> 
> * use `hub.config` where there is no explicit chart config (easiest for us)
> * or list the properties than can be set with `hub.config` (more work as we need to keep it up to date)

**Comment by minrk** [comment]
Yeah, I think in general, `hub.config` should link to the [jupyterhub api reference](https://jupyterhub.readthedocs.io/en/stable/reference/api/), which contains docs for configuration, and we should only cover what folks _shouldn't_ put in there. In general, I think _most_ config should be in this passthrough going forward, unless z2jh needs to handle it specially.

It may be relevant to also link to the config docs for each of the bundled packages, e.g. [kubespawner](https://jupyterhub-kubespawner.readthedocs.io/en/latest/spawner.html) and [oauthenticator](https://oauthenticator.readthedocs.io/en/latest/reference/api/index.html)
